### DISCUSSION TITLE
Invalid userset rewrite definition

### DISCUSSION THREAD

**Discussion by jdicami** [openfga/community#316] [Q&A]
Hello,

I have built openfga from source (latest main branch) so that I can use it as a library. I am encountering an issue when creating a new authorization model using the exact same definition as in the docs here https://openfga.dev/docs/getting-started/configure-model

When I call WriteAuthorizationModel from the server instance, it fails with "the definition of relation 'reader' in object type 'document' is invalid: invalid userset rewrite definition."

I have added the following print statements

```
	log.Println(string(encodedWriteAuthorizationModelRequest))
	log.Println(req.TypeDefinitions)
```

which prints out 
```
{"type_definitions":[{"type":"user"},{"relations":{"reader":{"this":{}}},"type":"document"}],"schema_version":"1.1","store_id":"01J04QJBFNGM82ZZP9Y1YW13VC"}

[type:"user" type:"document" relations:{key:"reader" value:{}}]
```

respectively. 

Any idea why`isUsersetRewriteValid` is failing.

**Comment by rhamzeh** [comment]
Hi @jdicami can you share a screenshot of where you found this json? I couldn't find it on the linked page. If it's copied from there verbatim then that would be an issue in the docs.

The JSON provided is missing the metadata needed that indicates what the type of the users related to document reader are.
